src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -254,8 +254,21 @@ public SamReader open(final SamInputResource resource) {
                                 validationStringency,
                                 this.samRecordFactory
                         );
+                    } else if (SamStreams.sourceLikeCram(data.asUnbufferedSeekableStream())) {
+                        if (referenceSource == null) {
+                            referenceSource = ReferenceSource.getDefaultCRAMReferenceSource();
+                        }
+                        SeekableStream bufferedIndexStream = indexDefined ?
+                                IOUtil.maybeBufferedSeekableStream(indexMaybe.asUnbufferedSeekableStream()) :
+                                null;
+                        primitiveSamReader = new CRAMFileReader(
+                                IOUtil.maybeBufferedSeekableStream(data.asUnbufferedSeekableStream()),
+                                bufferedIndexStream, referenceSource, validationStringency);
                     } else {
-                        throw new SAMFormatException("Unrecognized file format: " + data.asUnbufferedSeekableStream());
+                        // assume its a SAM file/no index
```

Comment:
Might want to emit a message on the logger when we fall into the assume SAM case